File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlToggle.cs
namespace T1.SqlSharp.Expressions;

public class SqlToggle : ISqlExpression
{
    public SqlType SqlType { get; } = SqlType.WithToggle;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_Toggle(this);
    }

    public string ToggleName { get; set; } = string.Empty;
    public string Value { get; set; } = string.Empty;


    public string ToSql()
    {
        return $"{ToggleName}={Value}";
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlToggle.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlToggle implements ISqlExpression {
    SqlType: SqlType = SqlType.WithToggle;
    Span: TextSpan = new TextSpan();

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_Toggle(this);
    }

    ToSql(): string {
        return '';
    }
} 